backend/src/controllers/items-controller.ts

```diff
@@ -67,9 +69,18 @@ export const createItem = async (req: CustomRequest, res: Response) => {
 
 
 //Retrieves a list of all items
-export const retrieveItems = async (req: Request, res: Response) => {
+export const retrieveItems = async (req: CustomRequest, res: Response) => {
     try {
-        const items = await Item.find();
+        //Checks if token is missing
+        if (!req.token) {
+            return res.status(400).json({ message: 'Token missing' })
+        }
+        const filter: any = {};
+        const groupId = req.query.groupId as string;
```

Comment:
(Repeat) Before adding the `groupId` to the `Item`, check that a `Group` object exists with its `_id` equal to `groupId`